elephant/spike_train_generation.py

```diff
@@ -1128,33 +1144,33 @@ def homogeneous_poisson_process_with_refr_period(rate,
     rate_mag = rate.rescale(1 / t_start.units).magnitude
     refr_period_mag = refr_period.rescale(t_start.units).magnitude
 
-    if rate_mag >= 1. / refr_period_mag:
-        raise ValueError(
-            'The refractory period implies an upper limit to the firing rate i.e. the firing rate if every spike is '
-            'separated exactly by the refractory period. This firing rate is the inverse of refractory period. '
-            'At the moment there is the refractory period: {}'.format(str(refr_period)) +
-            ' and the firing rate: {}. '.format(str(rate)) +
-            'One of these parameters need to be changed.'
-        )
+    if 1. / rate_mag <= refr_period_mag:
+        raise ValueError("Period between two successive spikes must be larger"
+                         "than the refractory period. Decrease either the"
+                         "firing rate or the refractory period.")
 
     duration = (t_stop - t_start).magnitude
-    mean_spike_count = rate_mag * duration
-    spike_count = np.random.poisson(lam=mean_spike_count)
-
-    # Check that the number of spikes drawn from the Poisson distribution,
-    # can fit in the duration regarding the refractory period.
-    if spike_count >= 1. + duration / refr_period_mag:
-        spike_count = np.ceil(duration / refr_period_mag).astype(int)
-
-    # Due to the refractory period the effective space in where the spikes can be placed is
-    # shortened by (spike-count - 1) times the refr. period.
-    eff_duration = duration - (spike_count - 1) * refr_period_mag
-
-    # In this effective time interval each spike is placed uniformly, and after sorting to each ISI
-    # one refractory period is added.
-    st = np.sort(np.random.random(spike_count)) * eff_duration
-    st += refr_period_mag * np.arange(spike_count)
-    st += t_start.magnitude
+    if duration <= refr_period:
+        st = []
```

Comment:
I would here also return a empty Spiketrain or an empty numpy-array. That it is consistent with the other output.

Reply:
yeah, then it'd require an additional if-check

Reply:
And a the case, that the duration is smaller than the refractory_period. You could still get one spike inside the spiketrain isn't it?

Reply:
hm, true